src/libraries/Microsoft.Extensions.Caching.Memory/tests/TokenExpirationTests.cs

```diff
@@ -207,6 +207,29 @@ public void TokenExpiresOnRegister()
             Assert.Null(result);
         }
 
+        [Fact]
+        public async Task PostEvictionCallbacksGetInvokedWhenMemoryCacheEntriesExpireWithAnActiveChangeToken()
+        {
+            using var cache = new MemoryCache(new MemoryCacheOptions());
+            var key = new object();
+
+            var cts = new CancellationTokenSource(TimeSpan.FromSeconds(1));
```

Comment:
Instead of doing this waiting, can we instead `cts.Cancel()` directly when we want to expire the entry?